lib/string.gi

```diff
@@ -1406,3 +1406,61 @@ InstallGlobalFunction(PrintFormatted, function(args...)
     # directed
     Print(CallFuncList(StringFormatted, args));
 end);
+
+InstallGlobalFunction(Pluralize,
+function(args...)
+  local nargs, str, len;
+
+  nargs := Length(args);
+  if not (nargs in [2, 3] and
+          IsInt(args[1]) and args[1] >= 0 and
+          ForAll([2 .. nargs], i -> IsString(args[i]))) then
+    ErrorNoReturn("Usage: Pluralize(<count>, <string>[, <plural>])");
+  elif args[1] = 1 then
+    return args[2];
+  elif nargs = 3 then
+    return args[3];
+  fi;
+  str := args[2];
+  len := Length(str);
+
+  # Special case weirdly short words
+  if len <= 1 then
+    return str;
+  fi;
+
+  # Guess and return the plural form of <str>.
+  # Inspired by the rails inflection rules
+
+  # Uncountable nouns
+  if str in ["equipment", "information", "series", "species"] then
+    return str;
+
+  # Irregular plurals
+  elif str = "axis" then
+    return "axes";
+  elif str = "child" then
+    return "children";
+  elif str = "person" then
+    return "people";
+
+  # Peculiar endings
+  elif str[len - 1] in "ie" and str[len] = 'x' then
+    return Concatenation(str{[1 .. len - 2]}, "ices");
+  elif len >= 3 and str{[len - 2 .. len]} = "sis" then
+    return Concatenation(str{[1 .. len - 3]}, "ses");
+  elif len >= 3 and str[len - 2] <> 'f' and str{[len - 1, len]} = "fe" then
```

Comment:
The following is perhaps less efficient but IMHO much clearer (and I am not worried about performance of this function)
```suggestion
  elif EndsWith( str, "fe" ) and not EndsWith( str, "ffe" ) then
```
I think almost all of these checks would be better to read if they used `EndsWith`. Clearly subjective, though!

Reply:
Agreed.